### PR TITLE
[EventHub] update checkpointstore service operations to take kwargs

### PR DESCRIPTION
**Pull request by swathipil**
fixes: #17154

### CODE REVIEW

**Comment by yunhaoling** [comment]
there're some pylint/mypy errors within the in_memory_checkpoint_store.py.

(for more context, this is an internal class which is used for test only, it's not part of the public api)

File: sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Release History
 
 ## 1.1.4 (Unreleased)
-
+- Updated `list_ownership`, `claim_ownership`, `update_checkpoint`, `list_checkpoints` on `BlobCheckpointStore` to support `**kwargs`.
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/CHANGELOG.md:4]: nit:
```suggestion
- Updated `list_ownership`, `claim_ownership`, `update_checkpoint`, `list_checkpoints` on `BlobCheckpointStore` to support taking `**kwargs`.
```

File: sdk/eventhub/azure-eventhub-checkpointstoreblob/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## 1.1.4 (Unreleased)
 
+**New features**
+- Updated `list_ownership`, `claim_ownership`, `update_checkpoint`, `list_checkpoints` on `BlobCheckpointStore` to support `**kwargs`.
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub-checkpointstoreblob/CHANGELOG.md:6]: nit:
```suggestion
- Updated `list_ownership`, `claim_ownership`, `update_checkpoint`, `list_checkpoints` on `BlobCheckpointStore` to support taking `**kwargs`.
```

File: sdk/eventhub/azure-eventhub/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## 5.3.2 (Unreleased)
 
+**New Features**
+
+- Updated `list_ownership`, `claim_ownership`, `update_checkpoint`, `list_checkpoints` on sync and async `CheckpointStore` to support `**kwargs`.
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub/CHANGELOG.md:7]: nit:
```suggestion
- Updated `list_ownership`, `claim_ownership`, `update_checkpoint`, `list_checkpoints` on sync and async `CheckpointStore` to support taking `**kwargs`.
```

**Comment by swathipil** [comment]
need to add a comment in CHANGELOG/address the following:
implementing `CustomCheckpointStore(CheckpointStore)` without methods that don't take kwargs results in the following pylint errors:
```
W0221: Parameters differ from overridden 'list_ownership' method (arguments-differ)
W0221: Parameters differ from overridden 'claim_ownership' method (arguments-differ)
W0221: Parameters differ from overridden 'update_checkpoint' method (arguments-differ)
W0221: Parameters differ from overridden 'list_checkpoints' method (arguments-differ)
```

File: sdk/eventhub/azure-eventhub/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,11 @@
 
 ## 5.3.2 (Unreleased)
 
+**New Features**
+
+- Updated `list_ownership`, `claim_ownership`, `update_checkpoint`, `list_checkpoints` on sync and async `CheckpointStore` to support taking `**kwargs`.
+- WARNING: Implementing a custom checkpointstore that does not support taking `**kwargs` in the methods listed previously will result in the following pylint error: `W0221: Parameters differ from overridden ________ method (arguments-differ)`.
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub/CHANGELOG.md:8]: nit: indentation?
```suggestion
  - WARNING: Implementing a custom checkpointstore that does not support taking `**kwargs` in the methods listed previously will result in the following pylint error: `W0221: Parameters differ from overridden ________ method (arguments-differ)`.
```

**Review by yunhaoling — APPROVED** [review]
LGTM

File: sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/partition_context.py [hunk 5]
```diff
@@ -78,7 +78,18 @@ def update_checkpoint(self, event=None):
...
                     "sequence_number": checkpoint_event.sequence_number,
                 }
-                self._checkpoint_store.update_checkpoint(checkpoint)
+                try:
+                    self._checkpoint_store.update_checkpoint(checkpoint, **kwargs)
+                except TypeError as e:
+                    if "update_checkpoint() got an unexpected keyword argument" in str(
+                        e
+                    ):
+                        _LOGGER.info(
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/partition_context.py:87]:
```suggestion
                        self._checkpoint_store.update_checkpoint(checkpoint)
                        _LOGGER.info(
```

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_eventprocessor/partition_context.py [hunk 6]
```diff
@@ -74,7 +74,7 @@ async def update_checkpoint(self, event: Optional["EventData"] = None) -> None:
                     "offset": checkpoint_event.offset,
                     "sequence_number": checkpoint_event.sequence_number,
                 }
-                await self._checkpoint_store.update_checkpoint(checkpoint)
+                await self._checkpoint_store.update_checkpoint(checkpoint, **kwargs)
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub/azure/eventhub/aio/_eventprocessor/partition_context.py:77]: the sync change needs to be applied here as well.
(my bad for forgetting about the async part in my previous review)

**Pull request merged**